vespa-http-client/src/main/java/com/yahoo/vespa/http/client/core/communication/DocumentQueue.java

```diff
@@ -38,10 +38,10 @@
         }
     }
 
-    void put(Document document) throws InterruptedException {
+    void put(Document document, boolean self) throws InterruptedException {
```

Comment:
It's not immediately obvious what the semantics of `self` are unless you look at the call site(s). Consider renaming to something more explicit, maybe `calledFromIoThread` or something like that.